File: stlm-java/src/segment/RootSegment.py
#!/usr/bin/env python
""" generated source for module RootSegment """
# package: segment
import sequence.Text

class RootSegment(InternalSegment):
    """ generated source for class RootSegment """
    text = Text()

    @overloaded
    def __init__(self, text):
        """ generated source for method __init__ """
        super(RootSegment, self).__init__(-1)
        self.text = text

    @__init__.register(object)
    def __init___0(self):
        """ generated source for method __init___0 """
        super(RootSegment, self).__init__()

    def getRight(self):
        """ generated source for method getRight """
        return 0

    def increment(self):
        """ generated source for method increment """

    def getCount(self):
        """ generated source for method getCount """
        return len(self.text)

File: stlm-java/src/segment/RootSegment.java
package segment;

import sequence.Text;

public class RootSegment<T> extends InternalSegment {
	
    public Text<T> text;
	
    public RootSegment(Text<T> text) {
    	super(-1);
    	this.text = text;
    }
    
    public RootSegment() {
    	
    }
     
    public int getRight() {
    	return 0;
    }
    
    public void increment() {
    	
    }
    
    public double getCount() {
    	return text.size();
    }


}
